homeassistant/components/group.py

```diff
@@ -161,48 +168,78 @@ def reload_service_handler(service_call):
     return True
 
 
-def _process_config(hass, config, component):
+@callback
+def _async_process_config(hass, config, component):
     """Process group configuration."""
+    groups = []
     for object_id, conf in config.get(DOMAIN, {}).items():
         name = conf.get(CONF_NAME, object_id)
         entity_ids = conf.get(CONF_ENTITIES) or []
         icon = conf.get(CONF_ICON)
         view = conf.get(CONF_VIEW)
 
-        group = Group(hass, name, entity_ids, icon=icon, view=view,
-                      object_id=object_id)
-        component.add_entities((group,))
+        group = Group.async_create_group(
+            hass, name, entity_ids, icon=icon, view=view, object_id=object_id)
+        groups.append(group)
 
-    return True
+    hass.loop.create_task(component.async_add_entities(groups))
```

Comment:
We can now no longer ensure that the groups have been added after setup is done. We should make this a coroutine. That will fix a lot of the tests 👍